src/views/controller.py

```diff
@@ -0,0 +1,21 @@
+from urwid import (
+    connect_signal,
+    disconnect_signal,
+    emit_signal,
+    register_signal,
+    )
+
+class Controller:
```

Comment:
imo ça devrait être une classe abstraite
https://docs.python.org/3/library/abc.html

Reply:
Traceback (most recent call last):
  File "./grader", line 3, in <module>
    from src.views.mail    import MailView
  File "/home/olivier/Documents/inf1900-grader/src/views/mail.py", line 4, in <module>
    from src.views.widgets import (
  File "/home/olivier/Documents/inf1900-grader/src/views/widgets.py", line 8, in <module>
    from src.views.buffer import Buffer, MiniBuffer
  File "/home/olivier/Documents/inf1900-grader/src/views/buffer.py", line 7, in <module>
    class Buffer(Edit, Controller):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases

Doesn't work

Reply:
Je crois que la classe Widget hérite déjà d'une classe meta abstraite. Il y a conflit

Reply:
ah, c'pas grave alors